doc/examples/xx_applications/plot_haar_extraction_selection_classification.py

```diff
@@ -0,0 +1,147 @@
+"""
+======================================================
+Face classification using Haar-like feature descriptor
+======================================================
+
+Haar-like feature descriptors were successfully used to implement the first
+real-time face detector [1]_. Inspired by this application, we propose an
+example illustrating the extraction, selection, and classification of Haar-like
+features to detect faces vs. non-faces.
+
+Notes
+-----
+
+This example relies on scikit-learn regarding the feature selection and
+classification.
+
+References
+----------
+
+.. [1] Viola, Paul, and Michael J. Jones. "Robust real-time face
+       detection." International journal of computer vision 57.2
+       (2004): 137-154.
+       http://www.merl.com/publications/docs/TR2004-043.pdf
+       DOI: 10.1109/CVPR.2001.990517
+
+"""
+from __future__ import print_function
+
+import sys
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import classification_report
+from sklearn.externals.joblib import Parallel, delayed
+
+from skimage.data import cbcl_face_database
+from skimage.transform import integral_image
+from skimage.feature import haar_like_feature
+from skimage.feature import haar_like_feature_coord
+from skimage.feature import draw_haar_like_feature
+
+print(__doc__)
+
+###############################################################################
+# The usual feature extraction scheme
+###############################################################################
+# The procedure to extract the Haar-like feature for an image is quite easy: a
+# region of interest (ROI) is defined for which all possible feature will be
+# extracted. The integral image of this ROI will be computed and all possible
+# features will be computed.
+
+
+def extract_feature_image(img, feature_type, feature_coord=None):
+    """Extract the haar feature for the current image"""
+    ii = integral_image(img)
+    return haar_like_feature(ii, 0, 0, ii.shape[0], ii.shape[1],
+                             feature_type=feature_type,
+                             feature_coord=feature_coord)
+
+
+# trick such that we can pickle this function when building the doc with
```

Comment:
Is this really necessary?  I asked about Dask before; does it have the same issue?

Reply:
I will check